src/x509/clu_x509_sign.c

```diff
@@ -469,6 +557,35 @@ int wolfCLU_CertSign(WOLFCLU_CERT_SIGN* csign, WOLFSSL_X509* x509, char* ext)
         }
     }
 
+    /* increment the serial number in the file */
+    if (ret == WOLFCLU_SUCCESS && csign->serialFile != NULL) {
+        WOLFSSL_ASN1_INTEGER* s;
+        char buf[EXTERNAL_SERIAL_SIZE*2];
+        int size = EXTERNAL_SERIAL_SIZE*2;
+
+        s = wolfSSL_ASN1_INTEGER_new();
+        wolfSSL_BIO_reset(csign->serialFile);
+        if (wolfSSL_a2i_ASN1_INTEGER(csign->serialFile, s, buf, size)
+                != WOLFSSL_SUCCESS) {
+            WOLFCLU_LOG(WOLFCLU_E0, "Issue reading serial number from file");
+            ret = WOLFCLU_FATAL_ERROR;
+        }
+
+        if (wolfSSL_ASN1_INTEGER_set(s, wolfSSL_ASN1_INTEGER_get(s)+1)
```

Comment:
What if `wolfSSL_ASN1_INTEGER_get` returns a negative? Perhaps setting this result to `long` and adding some checks might be useful?